test/unit/network/ConnectionRegistryTest.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { expect } = require('chai');
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+
+const { ConnectionRegistryImpl } = require('../../../lib/network/ConnectionManager');
+const { Connection } = require('../../../lib/network/Connection');
+const { ConnectionStrategyConfigImpl, ReconnectMode } = require('../../../lib/config/ConnectionStrategyConfig');
+const { RoundRobinLB } = require('../../../lib/util/RoundRobinLB');
+const { UuidUtil } = require('../../../lib/util/UuidUtil');
+const { ClientOfflineError, IOError } = require('../../../lib/core/HazelcastError');
+
+describe('ConnectionRegistryTest', function () {
+
+    const connectionState = {
+        INITIAL: 0,
+        CONNECTED_TO_CLUSTER: 1,
+        INITIALIZED_ON_CLUSTER: 2
+    };
+
+    afterEach(function () {
+        sandbox.restore();
+    });
+
+    describe('getRandomConnection', function () {
+        it('should use load balancer in smart mode', function () {
+            const loadBalancerStub = sandbox.stub(RoundRobinLB.prototype);
+            const connectionRegistry = new ConnectionRegistryImpl(
+                new ConnectionStrategyConfigImpl(),
+                true,
+                loadBalancerStub
+            );
+            connectionRegistry.getRandomConnection();
+            expect(loadBalancerStub.next.called).to.be.true;
+        });
+
+        it('should return first active connection in non-smart mode without using load balancer',
+        function () {
```

Comment:
The same nit applies to code below
```suggestion
            function () {
```